Name: Launcher Auto-Popup
Version: 1.3.5-1
Author: Jason Robitaille
Description: Will bring up launcher on system boot. Also, upon turning the screen on, if no cards are in focus, the launcher will pop up.  (MIT license)

--- .orig/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
+++ /usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
@@ -58,6 +58,10 @@
 		this.storageInit();
 		
 		//Initialize and Subscribe for Power Off Notification.
+		this.controller.serviceRequest('palm://com.palm.systemmanager', {
+			method: 'systemUi',
+			parameters: {launcher:true}
+		});
 		this.powerOffInit();
 		
 		//Subscribe to Powerd Notification.
@@ -2318,6 +2322,22 @@
 		},
 		onSuccess: this.powerOffHandleNotifications.bind(this)
 	});
+    this.displayChanged = this.controller.serviceRequest('palm://com.palm.display/control', {
+    		method:'status',
+ 		parameters:{
+ 			subscribe:true
+ 		},
+ 		onSuccess: function(response) {
+ 			if(response && response.event) {
+ 				if(response.event=="displayOn" && this.foregroundAppID==null) {
+ 					this.controller.serviceRequest('palm://com.palm.systemmanager', {
+						method: 'systemUi',
+						parameters: {launcher:true}
+					});
+ 				}
+ 			}
+ 		}.bind(this)
+	});
 },
 
 
